discourse: 3.4.7 -> 3.5.2

leona-ya opened the pull request:
https://meta.discourse.org/t/3-5-0-major-release/379212?tl=en
https://meta.discourse.org/t/3-5-1-security-and-maintenance-release/383511
https://meta.discourse.org/t/3-5-2-security-and-maintenance-release/386388

This release introduces a few rust-based ruby dependencies.

Multiple plugins were removed.
  - moved to discourse core in this discourse 3.5.0:
      - discourse-assign
      - discourse-calender
      - discourse-chat-integration
      - discourse-data-explorer
      - discourse-github
      - discourse-math
      - discourse-oauth2-basic
      - discourse-openid-connect
      - discourse-topic-voting
      - discourse-reactions
      - discourse-solved

  - moved to discourse core in earlier releases:
      - discourse-spoiler-alert

  - deprecated plugins:
     - discourse-canned-replies (since Aug. 2023)
     - discourse-migratepassword



<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

leona-ya committed:
discourse: 3.4.7 -> 3.5.1

https://meta.discourse.org/t/3-5-0-major-release/379212?tl=en
https://meta.discourse.org/t/3-5-1-security-and-maintenance-release/383511

This release introduces a few rust-based ruby dependencies.

Two plugins were removed
- discourse-migratepassword as it's deprecated
- discourse-topic-voting as it's integrated into discourse core

pgvector extension handling:
Co-Authored-By: talyz <kim.lindberger@gmail.com>

leona-ya commented:
I haven't tested this with a live instance.

mweinelt commented:
Impressive work!

Per https://meta.discourse.org/t/bundling-more-popular-plugins-with-discourse-core/373574?ref=blog.discourse.org I would have expected more plugins to have moved into core.

talyz commented on nixos/modules/services/web-apps/discourse.nix:
```diff
@@ -555,6 +555,17 @@ in
           + "Either update your PostgreSQL package to the correct version or set services.discourse.database.ignorePostgresqlVersion. "
           + "See https://nixos.org/manual/nixos/stable/index.html#module-postgresql for details on how to upgrade PostgreSQL.";
       }
+      {
+        assertion =
+          databaseActuallyCreateLocally
+          -> (builtins.elem config.services.postgresql.package.pkgs.pgvector (
+            # installedExtensions doesn't exist when when the postgresql package has no extensions
+            config.services.postgresql.package.installedExtensions or [ ]
+          ));
+        message =
+          "Discourse requires the PostgreSQL pgvector extension. Please extend services.postgresql.package accordingly. "
+          + "See https://nixos.org/manual/nixos/stable/index.html#module-postgresql for details.";
+      }
```
I don't think this should be necessary if we use the `extensions` option of the `postgresql` module. I have a working version here: https://github.com/talyz/nixpkgs/commit/1bc7000da42bc288e6130530514c76c8409ea0d4. Feel free to pull it in.

talyz commented on pkgs/servers/web-apps/discourse/default.nix:
This is already included in `runtimeDeps`.

talyz reviewed (changes requested):
I've tried this out a bit in our testing environment now, but rebased on master. It seems to work well! Great work!

`3.5.1` was released a few days ago: https://meta.discourse.org/t/3-5-1-security-and-maintenance-release/383511. Could you include it as well?

leona-ya committed:
discourse: remove deprecated and moved to core plugins

moved to discourse core in this discourse 3.5.0:
  - discourse-assign
  - discourse-calender
  - discourse-chat-integration
  - discourse-data-explorer
  - discourse-github
  - discourse-math
  - discourse-oauth2-basic
  - discourse-openid-connect
  - discourse-reactions
  - discourse-solved

moved to discourse core in earlier releases:
  - discourse-spoiler-alert

deprecated plugins:
  - discourse-canned-replies (since Aug. 2023)

leona-ya committed:
discourse.mkDiscoursePlugin: allow overriding pluginName

discourse.plugins.discourse-ldap-auth has the interesting situation that
it – as only package – only has `ldap` as name and not prefixed with
`discourse-`. I feel like having a `ldap` derivation without further
context is bad, so I went this way.

leona-ya committed:
discourse.plugins.discourse-ldap-auth: fix plugin name

leona-ya committed:
discourse: 3.5.1 -> 3.5.2

https://meta.discourse.org/t/3-5-2-security-and-maintenance-release/386388
